modelling/src/test/java/org/axonframework/modelling/entity/EntityCommandHandlingComponentTest.java

```diff
@@ -18,110 +18,214 @@
+                                                                                         "command");
+    private final MessageType mixedMessageType = new MessageType("MixedCommand");
+    private final CommandMessage<?> mixedCommandMessage = new GenericCommandMessage<>(mixedMessageType, "command");
+    private final String entityId = "myEntityId456";
+
     @Mock
     private Repository.LifecycleManagement<String, TestEntity> repository;
 
-    @Mock
-    private EntityModel<TestEntity> entityModel;
+    // We create a SimpleEntityModel, and rely on its API. If not, we basically have to mock every piece of behavior.
+    @Spy
+    private EntityModel<TestEntity> entityModel = EntityModel
+            .forEntityType(TestEntity.class)
+            .creationalCommandHandler(creationalMessageType.qualifiedName(),
+                                      ((command, c) -> resultMessage("creational")))
+            .creationalCommandHandler(mixedMessageType.qualifiedName(),
+                                      ((command, c) -> resultMessage("creational-mixed")))
+            .instanceCommandHandler(instanceMessageType.qualifiedName(), ((command, entity, c) -> resultMessage("instance")))
+            .instanceCommandHandler(mixedMessageType.qualifiedName(), ((command, entity, c) -> resultMessage("instance-mixed")))
+            .build();
 
     @Mock
     private EntityIdResolver<String> idResolver;
 
-    private final CommandMessage<?> commandMessage = mock(CommandMessage.class);
-    private final String entityId = "myEntityId456";
-    private final ManagedEntity<String, TestEntity> managedEntity = mock(ManagedEntity.class);
-    private final TestEntity mockEntity = mock(TestEntity.class);
-
+    @InjectMocks
     private EntityCommandHandlingComponent<String, TestEntity> testComponent;
 
     @BeforeEach
     void setUp() {
-        testComponent = new EntityCommandHandlingComponent<>(
-                repository,
-                entityModel,
-                idResolver
-        );
+        lenient().when(idResolver.resolve(any(), any())).thenReturn(entityId);
     }
 
     @Test
-    void returnsQualifiedNamesFromEntityModel() {
-        Set<QualifiedName> supportedCommands = Set.of(
-                new QualifiedName("command1"),
-                new QualifiedName("command3"),
-                new QualifiedName("command5")
-        );
-        when(entityModel.supportedCommands()).thenReturn(supportedCommands);
-
+    void returnsSupportedCommandFromModel() {
         Set<QualifiedName> result = testComponent.supportedCommands();
-        Assertions.assertEquals(supportedCommands, result);
+        Assertions.assertEquals(Set.of(
+                creationalMessageType.qualifiedName(),
+                instanceMessageType.qualifiedName(),
+                mixedMessageType.qualifiedName()
+        ), result);
     }
 
-    @Test
-    void loadsEntityBasedOnIdAndForwardsCommand() {
+    @Nested
+    class OnlyCreationalCommandHandler {
```

Comment:
I am missing a test for the `NoHandlerForCommandException` case.

Reply:
Adding case, although I will add it to the parent class, not in each individual nested test